File: benchmarks/julia/src/print_to_file.jl
using BenchmarkTools
using Test
using Printf

include("utils.jl")

function print_to_file(n, file_path)
    open(file_path, "w") do io
        for i in 1:n
            Printf.@printf(io, "%d %d\n", i, i + 1)
        end
    end
end

# Test output
print_to_file(3, "/tmp/test-file.txt")
@test open(f -> String(read(f)), "/tmp/test-file.txt", "r") == "1 2\n2 3\n3 4\n"

# Run benchmark
write_benchmark_result("print_to_file", @benchmark print_to_file(100_000, "/dev/null"))

File: benchmarks/javascript/src/print_to_file.js
// import { strict as assert } from "node:assert";
// import { benchmark, writeBenchmarkResult } from "./utils.js";
// import * as fs from "fs";

// const printToFile = (n) => {
//   let f = fs.openSync("/dev/null", "w");
//   for (let i = 1; i <= n; i++) {
//     fs.writeSync(f, `${i} ${i + 1}\n`);
//   }
//   fs.closeSync(f);
// };

// // Test output
// printToFile(3, "/test-file.txt");
// // let f = fs.openSync("/test-file.txt", "r");
// // console.log(fs.readFileSync());
// // assert(fs.readFileSync("/test-file.txt","utf8") == "1 2\n2 3\n3 4\n");
// // fs.closeSync(f);

// // Run benchmark
// let times = benchmark(printToFile, [100_000, "/dev/null"], 1000, 100);
// writeBenchmarkResult("print_to_file", times);
